### PR TITLE
Catches issue with 1.0.2 OpenSSL which leads to a fault.

### PR DESCRIPTION
**Pull request by imajes**
See issue #467 for more info.

@geemus - this is a prospective fix, happy to take a different approach if you have thoughts.

### CODE REVIEW

File: lib/excon/socket.rb [hunk 1]
```diff
@@ -241,6 +241,14 @@ def write_nonblock(data)
           # I wish that this API accepted a start position, then we wouldn't
           # have to slice data when there is a short write.
           written = @socket.write_nonblock(data)
+        rescue Errno::EFAULT
+          if OpenSSL::OPENSSL_LIBRARY_VERSION.split(' ')[1] == '1.0.2'
+            msg = "The version of OpenSSL this ruby is built against (1.0.2) has a vulnerability
+                   which causes a fault. For more, see https://github.com/excon/excon/issues/467"
+            raise SecurityError.new(msg)
```
Comment (geemus) [inline lib/excon/socket.rb]: I hadn't seen `SecurityError` before. I can find it in new Ruby docs, but how far back is it implemented? I just don't want to have a backwards incompatible thing, so StandardError or something might be safer (unless we know for sure SecurityError has been around for a while, definitely possible I just haven't run across it before).

**Comment by geemus** [comment]
Approach looks great, just some small concern around backwards compatibility if you could take a look.

File: lib/excon/socket.rb (hunk 1 above)
Comment (imajes) [inline lib/excon/socket.rb]: Yeah, i just looked up ruby standard exceptions and that one stood out. will do a quick bit of research to see how far back it goes. Excon supports 1.8.7+ ?

Comment (geemus) [inline lib/excon/socket.rb]: Yeah, for now at least we are still trying to support 1.8.7+. I imagine we'll start dropping old versions at some point, but we haven't really had particularly strong reasons to start down that path so far.

**Comment by imajes** [comment]
@geemus: seems ```SecurityError``` exists in 1.8.7: http://ruby-doc.org/core-1.8.7/SecurityError.html - sound good to go then?

**Comment by imajes** [comment]
@geemus: ping- just checking in on this to see if we can wrap it up?

**Comment by geemus** [comment]
Thanks for double checking, yeah seems great (just a little swamped today).

**Comment by geemus** [comment]
Fix out in 0.45.0

**Pull request merged**